packages/schematics/angular/guard/schema.json

```diff
@@ -47,6 +47,24 @@
       "type": "boolean",
       "default": false,
       "description": "When true, applies lint fixes after generating the guard."
+    },
+    "implements": {
+      "type": "array",
+      "description": "Specifies which interfaces to implement.",
+      "uniqueItems": true,
+      "items": {
+        "type": "string"
```

Comment:
This looks like an `enum` instead of general string.

Reply:
If I change this to `enum`, I get the following build error:

`Error: Invalid type enum in JSON Schema at Schema#/properties/implements/items`

I realized that the way this is being used the items could be updated to just a list of strings, thank you for pointing that out. Let me know what you think!

Reply:
Should be something like;

```js
    "implements": {
      "type": "array",
      "description": "Specifies which interfaces to implement.",
      "default": "none",
      "enum": [
           CanActivate",
          "CanActivateChild",
          "CanLoad"
      ]
    },
```

Reply:
@alan-agius4 I'm sorry I could be totally missing something (my first time working with json schema 😬), but the x prompt logic requires this array to be names as `items: []` as seen in https://github.com/angular/angular-cli/blob/master/packages/angular/cli/models/schematic-command.ts#L311

Are you saying in addition to the x-prompt defintion, also include the `enum` definition you describe here?

Thanks for your help btw! Just trying to learn and make sure I understand.

Reply:
Not really, I think it should be something like;

```
 "implements": {
      "type": "array",
      "description": "Specifies which interfaces to implement.",
      "default": "CanActivate",
      "enum": [
           "CanActivate",
           "CanActivateChild",
           "CanLoad"
      ],
      "x-prompt": {
        "message": "Which interfaces would you like to implement?",
        "type": "list",
        "multiselect": true,
        "items": [
          "CanActivate",
          "CanActivateChild",
          "CanLoad"
        ]
      }
}
...

Reply:
Okay that makes sense thank you!

Reply:
Note: updated the above is I spotted a mistake

Reply:
Note also that I have a followup planned to add auto-configuration of multi-select based on the schema.  It will eliminate the need to manually specify most of the prompt options and is the reason I originally suggested just using a string constraint for now (this schema would be adjusted in the process).

Reply:
@clydin would I be able to help out with the implementation of the multi select auto configuration?? I’d love to keep learning and contributing and this could transition well from the work I’ve done so far. Let me know what you think and thanks everyone for the help so far!

Reply:
@clydin @okeefem2 @alan-agius4 For info, the decision made here causes issues like this one: cyrilletuzi/vscode-angular-schematics#32. The x-prompts is a specific use of the schema, the classic schema properties should be respected for other tools to work (ie. this is an enum). For the same reason, `implements` should be flagged as `required`.